e2e_samples/parking_sensors/scripts/deploy_azdo_service_connections_azure.sh

```diff
@@ -30,52 +30,121 @@ set -o errexit
+    log "Service Principal App Object ID: $spnAppObjId"
+
+    # Get list of federated credentials
+    spnCredlist=$(az ad app federated-credential list --id "$spnAppObjId" --query "[].id" -o json)
+    log "Federated credentials: $spnCredlist"
+ 
+    credArray=($(echo "$spnCredlist" | jq -r '.[]'))
+    #(&& and ||) to log success or failure of each delete operation
+    for cred in "${credArray[@]}"; do
+        az ad app federated-credential delete --federated-credential-id "$cred" --id "$spnAppObjId" &&
+        log "Deleted federated credential: $cred" || 
+        log "Failed to delete federated credential: $cred"
+    done
+
+  # Refresh the list of federated credentials
+  spnCredlist=$(az ad app federated-credential list --id "$spnAppObjId" --query "[].id" -o json)
+  log "Federated credentials after deletion attempt: $spnCredlist"
+  if [ "$(echo "$spnCredlist" | jq -e '. | length > 0')" = "true" ]; then
+      log "Failed to delete federated credentials: $spnCredlist"
+      exit 1
+  fi
+  # Proceed to delete the service connection if no federated credentials remain
+  log "Hence, no federated credentials found for service principal: $spnAppObjId"
+
+  #Important:Giving time to the portal process the cleanup
+  log "Giving time to the portal process the cleanup..."
+  sleep 15
+
+  #Delete azdo service connection
+  delete_response=$(az devops service-endpoint delete --id "$sc_id" --project "$AZDO_PROJECT" --organization "$AZDO_ORGANIZATION_URL" -y )
+  if echo "$delete_response" | grep -q "TF400813"; then
+      log "Failed to delete service connection: $delete_response"
+      exit 1
+  fi
+  log "Successfully deleted service connection: $delete_response"
+
+fi
+
+# JSON config file
+cat <<EOF > ./devops.json
+{
+    "data": {
+      "subscriptionId": "$az_sub_id",
+      "subscriptionName": "$az_sub_name",
+      "creationMode": "Automatic",
+      "environment": "AzureCloud",
+      "scopeLevel": "Subscription"
+    },
+    "name": "$az_service_connection_name",
+    "type": "azurerm",
+    "url": "https://management.azure.com/",
+    "authorization": {
+      "scheme": "WorkloadIdentityFederation",
+      "parameters": {
+        "tenantid": "$TENANT_ID",
+        "scope": "/subscriptions/$az_sub_id/resourcegroups/$RESOURCE_GROUP_NAME"
+      }
+    },
+    "isShared": false,
+    "isReady": true,
+    "serviceEndpointProjectReferences": [
+        {
+          "description": "",
+          "name": "$az_service_connection_name",
+          "projectReference": {
+            "id": "$project_id",
+            "name": "$AZDO_PROJECT"
+          }
+        }
+    ]
+}
+EOF
+
+log "Create a new service connection"
+
+# Create the service connection using the Azure DevOps CLI
+response=$(az devops service-endpoint create --service-endpoint-configuration ./devops.json --org "$AZDO_ORGANIZATION_URL" -p "$AZDO_PROJECT")
+sc_id=$(echo "$response" | jq -r '.id')
+log "Created Connection: $response"
```

Comment:
```suggestion
log "Created Connection: $sc_id"
```
Logging the entire json response is not nice on the script output, I would suggest logging the id

Reply:
done